src/runner.rs

```diff
@@ -161,6 +162,17 @@ enum RunResult {
     NestedCall(FuncRef),
 }
 
+pub trait Loader {
+    fn load_function_body(&self, index: usize) -> Option<FuncBody>;
```

Comment:
This is returning the wasmi `FuncBody` which is private to the crate. Any other crate cannot construct a `Some` value here. I think we should use the `parity_wasm::elements::FuncBody` here.

Reply:
> This is returning the wasmi FuncBody which is private to the crate.

I know. My original intention was to provide a deserializing type that would implement the interface. Something like this:

```rust
struct DeserializingLoader<'a>(Fn(usize) -> &'a [u8]);

impl<'a> Loader for DeserializingLoader<'a> {
    fn load_function_body(&self, index: usize) -> Option<FuncBody> {
        let buffer = self.0(index);
        let mut reader = Cursor::new(buffer);
        let deserialized_body: elements::FuncBody = elements::FuncBody::deserialize(&mut reader).ok()?;

        Some(FuncBody {
            locals: deserialized_body.locals().to_vec(),
            code: todo!() // Compile the function to `isa::Instructions`
        })
    }
}
```

Reply:
Yeah I am in favor of using `parity_wasm::elements::FuncBody`. As is the code cannot be used outside of the crate.